src/core/Basics/Song.cpp

```diff
@@ -725,6 +726,53 @@ Song* SongReader::readSong( const QString& sFileName )
 	pSong->setPlaybackTrackEnabled( bPlaybackTrackEnabled );
 	pSong->setPlaybackTrackVolume( fPlaybackTrackVolume );
 	pSong->setActionMode( actionMode );
+	
+	// pan law
+	Sampler* pSampler = AudioEngine::get_instance()->get_sampler();
+	QString sPanLawType( LocalFileMng::readXmlString( songNode, "pan_law", "RATIO_STRAIGHT_POLYGONAL" ) );
```

Comment:
Ok, so how to match?
You mean defining something similar to an array of strings?
`panlawstrings[]`
and referring as `panlawstrings[ RATIO_STRAIGHT_POLYGONAL ]` for example?

Reply:
Since the `enum` can be used as an `int` you can do something like

```c++
int nPanLawType = LocalFileMng::readXmlInt( songNode, "pan_law_type", 0 );
pSong->setPanLawType( nPanLawType );
```

If you use an `enum class` this, however, is not possible and you have to map it's members to `int` or `QString`.

Anyway. It's not that important how to store the variable (as a string or a int) since a) the amount of memory it requires on the system of the user is negligible and b) it's just an internal serialization and not intended to be edited directly. You can also leave it as it is. I don't mind.

Reply:
I see. The reason why I would not save it as an int is because one could add another pan law type in the future (who knows? or for personal pan law tests) and the string keeps the file interpretation save.
I had in mind that writing/reading the string from/to the xml file could have be done in a safer way (I just have to go deeper in the `enum class` )

Reply:
Okay. Then just leave it as is. You don't have to rewrite it as `enum class`. (Why I brought that up the begin with: a number of `enum`s in the code where/are handled like `if ( m_nPanLawType == 5 )` instead of `if ( m_nPanLawType == LINEAR_CONST_SUM )` which makes the code harder to grasp in a totally unnecessary way. But your code hygiene is better, so don't mind :) )

Reply:
Ok!